### ISSUE TITLE
getTextZoom() on IOS returns 100 regardless of whatever text size has been selected from accessibility!

### ISSUE THREAD

**Issue by srameshr** [phonegap/phonegap-mobile-accessibility#29]
After my document ```load``` has been triggered I fire the ```getTextZoom``` method:
```
 MobileAccessibility.getTextZoom(function(zoom) {
  console.log(zoom); // This returns 100 regardless of system font accessibility size
});
```

**Comment by majornista** [comment]
Have you tried calling `MobileAccessibility.usePreferredTextZoom(true);` before calling `MobileAccessibility.getTextZoom` to make sure that your app subscribes to the OS level text scaling?

**Comment by srameshr** [comment]
Yeah tried it again. Still the same thing. It just passes the value ```100``` to the callback function.

**Comment by majornista** [comment]
I'm not able to replicate on iOS 9.2.1. Could you provide any further detail on your test environment?

**Comment by lobo87** [comment]
i fixed that by changing the source code of CDVMobileAccessibility.m
`#define iOS10Delta (([[[UIDevice currentDevice] systemVersion] floatValue] >= 10.0 ) ? 40 : 0 )
-(double) mGetFontScale
{
    double fontScale = 1;
    if (iOS7Delta)  {
        fontScale = [[UIFont preferredFontForTextStyle:UIFontTextStyleBody] pointSize] / BASE_UI_FONT_TEXT_STYLE_BODY_POINT_SIZE;
    }
    
    if(iOS10Delta) {
        fontScale = [[UIFont preferredFontForTextStyle:UIFontTextStyleBody] pointSize] / 17;
    }
    
    return fontScale;
}`

this is propably not the nicest solution, because im not really sure for what the scaling value in the #define part is for, but i added it anyway

**Comment by ostigley** [comment]
For anyone else coming across this now, I think the getTextZoom function is now returning a promise, and does not take a callback as indicated in the documentation.  MY implementation seems to be working on IOS and Android:

```js
 const mBA = this.mobileAccessibility;
      mBA.getTextZoom()
        .then(textZoom => {
          // if the device zoom is between 100% and 125%, then use it
          if (100 < textZoom && textZoom < 126) {
            mBA.usePreferredTextZoom(true);
          } else if (textZoom > 125) {
            // if the device zoom is greater than 125%, then ignore it and use 125%, otherwise our designs will break
            mBA.usePreferredTextZoom(false);
            mBA.setTextZoom(125);
          } else {
            // if the device zoom is less than 100%, then ignore it and use 100%, otherwise our designs will break
            mBA.usePreferredTextZoom(false);
            mBA.setTextZoom(100);
          }
```

**Comment by m-saifuddin** [comment]
Dear @ostigley ,

Can you tell me exactly where I use your suggested code ? so solve the issue.

**Comment by m-saifuddin** [comment]
Dear,
Firstly Good Work!!.

This plugin works fine in Android but I am facing the same issue.. in iOS getTextZoom function return me 100 value even changing the text size in accessibility.
